src/mento.test.ts

```diff
@@ -348,124 +583,360 @@ describe('Mento', () => {
+                address: tokenOut,
+                symbol: fakeSymbolsByTokenAddr[tokenOut],
+              },
+            ],
+            path: [
+              {
+                providerAddr: mockedProvider,
+                id: exchange.exchangeId,
+                assets: [tokenIn, tokenOut],
+              },
+            ],
+          }
+
+          const fakeTxObj = { to: '0x1337', data: '0x345' }
+          const fakePopulatedTxObj = {
+            to: '0x123',
+            data: '0x00456',
+            from: '0xad3',
+            gasLimit: 2200,
+          }
+
+          mockBroker.populateTransaction.swapOut.mockReturnValueOnce(fakeTxObj)
+          const spy = jest
+            .spyOn(signer, 'populateTransaction')
+            // @ts-ignore
+            .mockReturnValueOnce(fakePopulatedTxObj)
+
+          const result = await testee.swapOut(
+            tokenIn,
+            tokenOut,
+            oneInWei,
+            twoInWei,
+            directPair
+          )
+          expect(result).toBe(fakePopulatedTxObj)
+
+          expect(mockBroker.populateTransaction.swapOut).toHaveBeenCalledWith(
+            mockedProvider,
+            exchange.exchangeId,
+            tokenIn,
+            tokenOut,
+            oneInWei,
+            twoInWei
+          )
+
+          expect(spy).toHaveBeenCalledWith(fakeTxObj)
+        }
+      }
     })
 
-    it('should throw if no exchange is found for the given tokens', async () => {
+    it('should call router.swapOut with the right parameters for a routed swap', async () => {
       const testee = await Mento.create(signer)
-      const tokenIn = fakeCeloUSDExchange.assets[0]
-      const tokenOut = 'fakeAsset'
+      const routedPair: TradablePair = {
+        id: 'cUSD-cEUR',
+        assets: [
+          {
+            address: fakecUSDTokenAddr,
+            symbol: fakeSymbolsByTokenAddr[fakecUSDTokenAddr],
+          },
+          {
+            address: fakecEURTokenAddr,
+            symbol: fakeSymbolsByTokenAddr[fakecEURTokenAddr],
+          },
+        ],
+        path: [
+          {
+            providerAddr: fakeUsdAndEurExchangeProvider,
+            id: fakeCeloUSDExchange.exchangeId,
+            assets: [fakecUSDTokenAddr, fakeCeloTokenAddr],
+          },
+          {
+            providerAddr: fakeUsdAndEurExchangeProvider,
+            id: fakeCeloEURExchange.exchangeId,
+            assets: [fakecEURTokenAddr, fakeCeloTokenAddr],
+          },
+        ],
+      }
 
-      await expect(
-        testee.swapOut(tokenIn, tokenOut, oneInWei, oneInWei)
-      ).rejects.toThrow(`No exchange found for ${tokenIn} and ${tokenOut}`)
+      const fakeTxObj = { to: '0x1337', data: '0x345' }
+      const fakePopulatedTxObj = {
+        to: '0x123',
+        data: '0x00456',
+        from: '0xad3',
+        gasLimit: 2200,
+      }
+
+      mockRouter.populateTransaction.swapTokensForExactTokens.mockReturnValueOnce(
+        fakeTxObj
+      )
+      const spy = jest
```

Comment:
no need to assign this to the `spy` constant as it's not being used anywhere that I can see

```suggestion
      jest
```